python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,417 @@
+    """ATCalsys usages definition.
+
+    Notes
+    -----
+
+    Additional usages definition:
+
+    * Setup: Enable ATWhitelight and perform warmup. Enable
+             ATMonochromator, Electrometer, and FiberSpectrographs.
+    * Configure: Adjust ATWhitelight and ATMonochromator for type of flat.
+    * TakeFlat: Take flatfield image with LATISS, Electrometer and FiberSpec
+    * DryTest: Disable CSCs and unit tests
+    """
+
+    Setup = 1 << 3
+    Configure = 1 << 4
+    TakeFlat = 1 << 5
+
+    def __iter__(self) -> typing.Iterator[int]:
+        return iter(
+            [
+                self.All,
+                self.StateTransition,
+                self.MonitorState,
+                self.MonitorHeartBeat,
+                self.Setup,
+                self.Configure,
+                self.TakeFlat,
+                self.DryTest,
+            ]
+        )
+
+
+class ATCalsys(BaseCalsys):
+    """LSST Auxiliary Telescope Calibration System.
+
+    ATCalsys encapsulates core functionality from the following CSCs: ATCamera,
+    ATWhiteLight, ATMonochromoter, Fiberspectrograph, Electrometer.
+
+    Parameters
+    ----------
+    domain : `lsst.ts.salobj.Domain`
+        Domain for remotes. If `None` create a domain.
+    log : `logging.Logger`
+        Optional logging class to be used for logging operations. If `None`,
+        creates a new logger. Useful to use in salobj.BaseScript and allow
+        logging in the class use the script logging.
+    intended_usage: `int`
+        Optional integer that maps to a list of intended operations. This is
+        used to limit the resources allocated by the class by gathering some
+        knowledge about the usage intention. By default allocates all
+        resources.
+    """
+
+    def __init__(
+        self,
+        domain: typing.Optional[salobj.Domain] = None,
+        log: typing.Optional[logging.Logger] = None,
+        intended_usage: typing.Optional[int] = None,
+        atcamera: typing.Optional[ATCamera] = None,
+    ) -> None:
+
+        self.electrometer_index = 1
+        self.fiberspectrograph_index = 1
+
+        super().__init__(
+            components=[
+                "ATMonochromator",
+                "ATWhiteLight",
+                f"FiberSpectrograph:{self.fiberspectrograph_index}",
+                f"Electrometer:{self.electrometer_index}",
+            ],
+            instrument_setup_attributes=[
+                "filter",
+                "calib_type",
+                "grating",
+                "entrance_slit",
+                "exit_slit",
+            ],
+            domain=domain,
+            log=log,
+            intended_usage=intended_usage,
+        )
+
+        self.atcamera = atcamera
+
+        self.entrance_slit = 0.5
+        self.exit_slit = 0.5
+
+        self.whitelight_startup_power = 1000
+
+        self.chiller_temp_tolerance_relative = 0.2
+        self.chiller_temperature = 20
+
+        self.wavelength = float(500)  # nm
```

Comment:
I have it here because you could run `configure_flat` without setting a wavelength and it will send an error because there is no self.wavelength. If you are running white light flats, it doesn't matter what self.wavelength is set to